drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -68,6 +68,87 @@ local device_type_profile_map = {
 }
 local detect_matter_thing
 
+local embedded_cluster_utils = require "embedded-cluster-utils"
+
+local CUMULATIVE_ENERGY_IS_REPORTED = "CUMULATIVE_ENERGY_IS_REPORTED"
+local TOTAL_EXPORTED_ENERGY = "TOTAL_EXPORTED_ENERGY"
+local LATEST_EXPORTED_TIMESTAMP = "LATEST_EXPORTED_TIMESTAMP"
+local RECURRING_EXPORT_REPORT_POLL_TIMER = "RECURRING_EXPORT_REPORT_POLL_TIMER"
+
+local REPORT_SCHEDULE_INIT_CALLED = "REPORT_SCHEDULE_INIT_CALLED"
+local REPORT_POLL_TIMER_IS_SET = "REPORT_POLL_TIMER_IS_SET"
+local RECURRING_POLL_TIMER = "RECURRING_POLL_TIMER"
+local REPORT_TIMEOUT = "REPORT_TIMEOUT"
+local MINIMUM_ST_ENERGY_REPORT_INTERVAL = (15 * 60) -- 15 minutes, reported in seconds
+local SUBSCRIPTIONS_COMPLETE = "SUBSCRIPTIONS_COMPLETE"
+
+local POWER_TOPOLOGY_TYPE = "POWER_TOPOLOGY_TYPE"
+local POWER_TOPOLOGY_FEATURE_MAP = {
+  ["NODE"] = 0x0001,
+  ["TREE"] = 0x0010,
+  ["SET"]  = 0x0100,
+  ["DYPF"] = 0x1100,
+}
+
+-- Return a ISO 8061 formatted timestamp in UTC (Z)
+-- @return e.g. 2022-02-02T08:00:00Z
+local function iso8061Timestamp(time)
+  return os.date("!%Y-%m-%dT%TZ", time)
+end
+
+local function delete_poll_schedule(device)
+  local poll_timer = device:get_field(RECURRING_POLL_TIMER)
+  if poll_timer then
+    device.thread:cancel_timer(poll_timer)
+    device:set_field(RECURRING_POLL_TIMER, nil)
+  end
+end
+
+local function send_poll_report(device, LAST_REPORT_TIME, TOTAL_ENERGY)
```

Comment:
`LAST_REPORT_TIME` and `TOTAL_ENERGY` do not need to be parameters since they are keys on the device object and the only instances in which `send_poll_report` is called, it always uses the same values for these parameters.

Reply:
Yeah, the thing is I built this when I was under the false impression that we were implementing both imported and exported energy, and this was meant to generalize. I think it should probably be removed for now, but it seems like partners are keen on using the imported energy feature, and so this may be useful again soon.